Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs

```diff
@@ -328,41 +348,41 @@ void IAccessibilityElementsController.ResetAccessibilityElements()
 
 		void UpdateUseSafeArea()
 		{
-			if (!Forms.IsiOS11OrNewer)
+			if (Element == null)
 				return;
 
-			if (!UsingSafeArea)
+			var tabThickness = _tabThickness;
+			if (!_isInItems)
+				tabThickness = 0;
+
+			if (Forms.IsiOS11OrNewer)
 			{
-				var safeAreaInsets = SafeAreaInsets;
-				if (safeAreaInsets == Page.Padding)
-					Page.Padding = default(Thickness);
+				var insets = NativeView.SafeAreaInsets;
+				if (Page.Parent is TabbedPage)
+				{
+					insets.Bottom = 0;
+				}
+
+				Page.On<PlatformConfiguration.iOS>().SetSafeAreaInsets(new Thickness(insets.Left, insets.Top + tabThickness, insets.Right, insets.Bottom));
 			}
-			else
+			else if (IsPartOfShell)
 			{
-				Page.Padding = SafeAreaInsets;
+				Page.On<PlatformConfiguration.iOS>().SetSafeAreaInsets(new Thickness(0, TopLayoutGuide.Length + tabThickness, 0, BottomLayoutGuide.Length));
 			}
-		}
 
-		void UpdateShellInsetPadding()
-		{
-			if (!(Element?.Parent is ShellContent))
-				return;
+			bool usingSafeArea = Page.On<PlatformConfiguration.iOS>().UsingSafeArea();
+			bool isSafeAreaSet = Element.IsSet(PageSpecific.UseSafeAreaProperty);
 
-			nfloat topPadding = 0;
-			nfloat bottomPadding = 0;
+			if (IsPartOfShell && !isSafeAreaSet)
```

Comment:
So the default behavior for `Page` is to have `UseSafeArea` default to `false`?  Why would this behave differently?  Is this going to effect more than just the root page of `ShellSection`?

Reply:
Looking at your initial issue. The problem is that shell just flat out ignored the SafeArea setting. With this PR your code here https://github.com/seansparkman/ShellNavigationExample/blob/use-safe-area-ignored/ShellNavigationRegistration/ShellNavigationRegistration/Views/OnBoarding.xaml#L13 will cause usingSafeArea to be false

https://github.com/xamarin/Xamarin.Forms/pull/6457/files#diff-d95827e5633c20beb139e10ccd1a40d9R415 so whatever you set the padding to is going to be the padding used for the page

Reply:
Right, but does `ContentPage` that isn't a part of `Shell` normally use the SafeAreaInsets?  Why would it behave differently for `Shell` and does this change effect all pages or just the first one?

Reply:
Good point!! I added an additional check so if there's nothing up top then it should just take the pages full height